File: project/BBWT.Client/src/app/main/aggregated-logs/log.ts
export interface ILog {
    id: number;
    message: string;
    level: string;
    timeStamp: Date;
    exception: string;
    logEvent: string;
    appName: string;
    server: string;
    iP: string;
    source: string;
    userName: string;
    isImpersonating?: boolean;
    originaluserName: string;
    errorId: string;
    httpStatus?: number;
}

export enum TimeWindow {
    TenMinutes = 0,
    ThirtyMinutes = 1,
    OneHour = 2,
    ThreeHours = 3,
    TwelveHours = 4
  }

File: modules/BBWM.AggregatedLogs/Log.cs
using BBWM.Core.Data;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace BBWM.AggregatedLogs
{
    [Index(nameof(Level))]
    [Index(nameof(TimeStamp))]
    public class Log : IEntity
    {
        public int Id { get; set; }

        [MaxLength(2000)]
        public string Message { get; set; }


        [MaxLength(20)]
        public string Level { get; set; }

        public DateTimeOffset TimeStamp { get; set; }

        public string Exception { get; set; }

        [MaxLength(2000)]
        public string LogEvent { get; set; }

        [MaxLength(100)]
        public string AppName { get; set; }

        [MaxLength(500)]
        public string Server { get; set; }

        [MaxLength(20)]
        public string IP { get; set; }

        [MaxLength(100)]
        public string Source { get; set; }

        [MaxLength(50)]
        public string UserName { get; set; }

        public bool? IsImpersonating { get; set; }

        [MaxLength(50)]
        public string OriginalUserName { get; set; }

        [MaxLength(50)]
        public string ErrorId { get; set; }

        public int? HttpStatus { get; set; }
    }
}
